packages/react/src/pages/crm-contact-list/contact-panel/contactPanel.tsx

```diff
@@ -0,0 +1,225 @@
+
+import React, { useEffect, useState, useCallback } from 'react';
+import { useNavigate } from 'react-router-dom';
+import classNames from 'classnames';
+import './contactPanel.scss';
+import { Button, DropDownButton, ScrollView } from 'devextreme-react';
+import Toolbar, { Item as ToolbarItem } from 'devextreme-react/toolbar';
+import Form, { Item as FormItem, ColCountByScreen } from 'devextreme-react/form';
+import Accordion, { Item as AccordionItem } from 'devextreme-react/accordion';
+import LoadPanel from 'devextreme-react/load-panel';
+import { ClickEvent as ButtonClickEvent } from 'devextreme/ui/button';
+import { formatNumber } from 'devextreme/localization';
+import { getContact } from 'dx-template-gallery-data';
+import { Contact } from '../../../shared/types/crm-contact';
+import { CardActivities } from '../../../components/card-activities/CardActivities';
+import { FormTextbox, FormPhoto, ContactStatus } from '../../../components';
+
+export const ContactPanel = ({ contactId, isOpen, changePanelOpen } : { contactId: number, isOpen: boolean, changePanelOpen:()=> void }) => {
+
+  const [contact, setContact] = useState<Contact>();
+  const [loading, setLoading] = useState(false);
+  const [isPin, setIsPin] = useState(false);
+  const [isEditing, setIsEditing] = useState(false);
+
+  const navigate = useNavigate();
+
+  const loadData = useCallback(() => {
+    setLoading(true);
+    getContact(contactId)
+      .then((data) => {
+        setContact(data);
+        setLoading(false);
+      })
+      .catch((error) => console.log(error));
+  }, [contactId]);
+
+  useEffect(() => {
+    if(contactId) {
+      loadData();
+    }
+  }, [loadData]);
+
+  const formatPrice = (price) => {
+    return formatNumber(price, {
+      type: 'currency',
+      currency: 'USD',
+    });
+  };
+
+  const onDataChanged = useCallback(data => {
+    setContact(data);
+  }, []);
+
+  const updateField = (field: string) => (value) => {
+    onDataChanged({ ...contact, ...{ [field]: value } });
+  };
+
+  const pinClick = useCallback(() => {
+    setIsPin(!isPin);
+  }, [isPin]);
+
+  const closePanel = useCallback(() => {
+    setIsPin(false);
+    changePanelOpen();
+  }, []);
+
+  const toggleEdit = useCallback(() => {
+    setIsEditing(!isEditing);
+  }, [isEditing]);
+
+  const navigateToDetails = () => {
+    navigate('/crm-contact-details');
+  };
+
+  const accordionTitleClick = (e: ButtonClickEvent) => {
+    e.event?.stopPropagation();
+  };
+
+  const customTitle = (item) => {
```

Comment:
Some of the methods has not useCallback. Some of the methods does not set state, they should move out of functional component